src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -1,37 +1,37 @@
 (ns puppetlabs.services.master.master-core
   (:import (java.io FileInputStream))
-  (:require [compojure.core :as compojure]
-            [me.raynes.fs :as fs]
-            [puppetlabs.puppetserver.ringutils :as ringutils]))
+  (:require [me.raynes.fs :as fs]
+            [puppetlabs.puppetserver.ringutils :as ringutils]
+            [puppetlabs.comidi :as comidi]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Routing
 
 (defn v2_0-routes
-  "Creates the compojure routes to handle the master's '/v2.0' routes."
+  "Creates the web routes to handle the master's '/v2.0' routes."
   [request-handler]
-  (compojure/routes
-    (compojure/GET "/environments" request
+  (comidi/routes
+    (comidi/GET "/environments" request
                    (request-handler request))))
 
 (defn legacy-routes
-  "Creates the compojure routes to handle the master's 'legacy' routes
+  "Creates the web routes to handle the master's 'legacy' routes
    - ie, any route without a version in its path (eg, /v2.0/whatever) - but
    excluding the CA-related endpoints, which are handled separately by the
    CA service."
   [request-handler]
-  (compojure/routes
-    (compojure/GET "/node/*" request
+  (comidi/routes
+    (comidi/GET ["/node/" :node] request
                    (request-handler request))
-    (compojure/GET "/facts/*" request
+    (comidi/GET ["/facts/" :node] request
```

Comment:
One difference that's going to happen here and for the `/node` endpoint is that a request to e.g. `/production/node/foo/bar` with this code will return a 404 not found from puppet server, whereas a with the previous code it would match at puppet server's level, and then get passed down to the indirector code, which will return `Could not retrieve facts for foo/bar: invalid key`. I'm not sure that this is at all an issue, but it is a change in behavior that I wanted to point out.